modules/labs-react/text-input/lib/hooks/useTextInputModel.tsx

```diff
@@ -0,0 +1,45 @@
+import {
+  createEventMap,
+  Model,
+  ToModelConfig,
+  useEventMap,
+  useUniqueId,
+} from '@workday/canvas-kit-react/common';
+
+type TextInputState = {
+  hasError?: boolean;
+  inputId?: string;
+  hintId?: string;
+  isRequired?: boolean;
+};
+
+type TextInputEvents = {};
+
+export type TextInputModel = Model<TextInputState, TextInputEvents>;
+
+const textInputEventMap = createEventMap<TextInputEvents>()({
+  guards: {},
+  callbacks: {},
+});
+
+export type TextInputModelConfig = TextInputState &
+  Partial<ToModelConfig<TextInputState, TextInputEvents, typeof textInputEventMap>>;
+
+export const useTextInputModel = (config: TextInputModelConfig = {}): TextInputModel => {
```

Comment:
Something to consider. All these models are field-type models. Will there be a specific model for each field type that is copy/pasted from this one, or is there a `FieldModel` instead of `*Model`? Popups work this way where all popup type components use a `PopupModel` instead of defining new model types for specific popups. `useModalModel` actually returns a `PopupModel` that has modal behaviors already baked in.